google/cloud/bigtable/table.py

```diff
@@ -782,6 +783,177 @@ def mutations_batcher(self, flush_count=FLUSH_COUNT, max_row_bytes=MAX_ROW_BYTES
+                -  ``state:READY`` --> The Backup is created and ready for use.
+                -  ``(name:howl) AND (start_time < \"2020-05-28T14:50:00Z\")``
+                		--> The Backup name contains the string "howl" and
+                		the Backup start time is before 2020-05-28T14:50:00Z.
+                -  ``size_bytes > 10000000000`` --> The Backup size is greater
+                		than 10GB
+
+        :type order_by: str
+        :param order_by: (Optional) An expression for specifying the sort order
+        				 of the results of the request. The string value should
+        				 specify one or more fields in ``Backup``. The full
+        				 syntax is described at https://aip.dev/132#ordering.
+
+                		 Fields supported are: \* name \* source\_table \*
+                		 expire\_time \* start\_time \* end\_time \*
+                		 size\_bytes \* state
+
+                		 For example, "start\_time". The default sorting order
+                		 is ascending. To specify descending order for the
+                		 field, a suffix " desc" should be appended to the
+                		 field name. For example, "start\_time desc". Redundant
+                		 space characters in the syntax are insigificant. If
+                		 order_by is empty, results will be sorted by
+                		 ``start_time`` in descending order starting from
+                		 the most recently created backup.
+
+        :type page_size: int
+        :param page_size: (Optional) The maximum number of resources contained
+        				  in the underlying API response. If page streaming is
+        				  performed per-resource, this parameter does not
+        				  affect the return value. If page streaming is
+        				  performed per-page, this determines the maximum
+        				  number of resources in a page.
+
+        :rtype: :class:`~google.api_core.page_iterator.Iterator`
+        :returns: Iterator of :class:`~google.cloud.bigtable.backup.Backup`
+                  resources within the current Instance.
+        :raises: :class:`ValueError <exceptions.ValueError>` if one of the
+                 returned Backups' name is not of the expected format.
+        """
+        cluster_id = cluster_id or "-"
+
+        backup_list_pb = self._instance._client.table_admin_client.list_backups(
+            self._instance.name + "/clusters/" + cluster_id,
+            filter_=filter_,
+            order_by=order_by,
+            page_size=page_size,
+        )
+
+        result = []
+        for backup_pb in backup_list_pb:
+            backup_id = backup_pb.name.split("/")[-1]
+            backup_cluster_id = backup_pb.name.split("/")[-3]
+            backup_expire_time = backup_pb.expire_time
+            result.append(self.backup(
+                backup_id,
+                cluster_id=backup_cluster_id,
+                expire_time=backup_expire_time
+            ))
+
+        return result
```

Comment:
Suggesting re-using `backup.from_pb` method here which also verifies that backup_pb.name is as expected and sets other values `start_time`, `end_time`, `size_bytes`, `state` as well.

```py
return [Backup.from_bp(backup_pb, self) for backup_pb in backup_list_pb]
```

Reply:
Also suggesting to add `list_backups` to the `Instance` class as well to list backups for given instance as well.
If yes, then this method should delegate to the `Instance.list_backups`.

Reply:
> Suggesting re-using `backup.from_pb` method here which also verifies that backup_pb.name is as expected and sets other values `start_time`, `end_time`, `size_bytes`, `state` as well.
> 
> ```python
> return [Backup.from_bp(backup_pb, self) for backup_pb in backup_list_pb]
> ```

Scratch above

Since `gapic.list_backups` returns a `PageIterator`, let's return an iterator to the user as well, without cashing all the data.

introduce a new method:

```python
def _item_to_backup(self, iterator, backup_pb):
    # Convert a backup protobuf to a Backup object.

    return Backup.from_pb(backup_pb, self)
```
use the above method to set the `item_to_value` property on the `page_iter` returned by `gapic.list_backups` to be used on [`next`](https://github.com/googleapis/python-api-core/blob/master/google/api_core/page_iterator.py#L133)
```python
  backup_page_iter_pb = self._instance._client.table_admin_client.list_backups(
      self._instance.name + "/clusters/" + cluster_id,
      filter_=filter_,
      order_by=order_by,
      page_size=page_size,
  )

  backup_page_iter_pb.item_to_value = self._item_to_backup

  return page_iter
```
WDYT?